content/blog/2020/12/2020-12-16-jira-upgrade-for-the-jenkins-project.adoc

```diff
@@ -0,0 +1,134 @@
+
+We're grateful for the long-standing contribution that Atlassian provides by donating the Jira license to the Jenkins project.
+We're grateful to the link:https://osuosl.org/[Oregon State University Open Source Lab] for their donation of equipment and bandwidth to host the server.
+
+== Upgrade Timeline
+
+We were running Jira 7.13 and had been managing that installation for a few years.
+Atlassian announced that Jira 7.13 would end its support life on November 28, 2020.
+We needed to upgrade from Jira 7.13 to a more recent version of Jira.
+As part of our membership in the Continuous Delivery Foundation, a Linux Foundation initiative, we could use their project services team to manage our Jira server.
+We decided to move from hosting our own Jira server to having the Jira experts at the Linux Foundation host it.
+
+The upgrade timeline looked like this:
+
+* November 2019 - Infrastructure team begins discussions about the November 2020 end of support for Jira 7.13
+* August 2020 - First conversations with Linux Foundation to host Jira for the Jenkins project.  Draft of the upgrade plan assembled and shared with the community
+* September 2020 - Schedule for testing week and final transition week proposed.  Authentication options evaluated and selected
+* October 2020 - Test upgrade performed and tested
+* November 2020 - Final upgrade completed and verified
+
+== Confronting the Complications
+
+Initial discussions between the Jenkins infrastructure team and the Linux Foundation identified complications related to authentication and SSL certificates.
+We planned, negotiated, and tested our assumptions throughout the project.
+
+=== Authentication
+
+Jira servers at the Linux Foundation typically use Linux Foundation accounts for user access.
+Unfortunately, the Jenkins LDAP database includes over 100,000 users and for many of them, Linux Foundation username doesn't correspond to Jenkins account username.
+It was not feasible to transition 100,000 user accounts from the Jenkins LDAP database to the Linux Foundation accounts system and still complete the Jira upgrade before the November 28, 2020 deadline.
+
+The Linux Foundation Project Services team evaluated authentication alternatives and confirmed that they could use the Jenkins LDAP server.
+Using the Jenkins LDAP server spared us from two transitions, LDAP and Jira, and kept the project timeline feasible.
+
+=== SSL Certificates
+
+Jira servers at the Linux Foundation use Let's Encrypt to generate SSL certificates for HTTPS.
+The Linux Foundation uses the DNS method to obtain SSL certificates.
+Unfortunately, the Jenkins project uses the HTTP method to obtain SSL certificates.
+
+Thankfully, Olivier Vernin of the Jenkins project and Anton Baranov of the Linux Foundation found a solution.
+They were able to grant sufficient permissions in the Jenkins DNS records so that the Linux Foundation could issue SSL certificates for issues.jenkins.io and issues.jenkins-ci.org.
```

Comment:
Correct DNS phrasing based on feedback from @olblak 

@olblak does this express it more accurately?

```suggestion
They created an ACME record in the Jenkins DNS server and pointed the issues.jenkins.io DNS record at the new Linux Foundation Jira server.
```